include/circt/Dialect/SV/SVDialect.h

```diff
@@ -45,7 +45,10 @@ StringRef legalizeName(llvm::StringRef name,
 ///
 /// Call \c legalizeName() to obtain a legal version of the name.  If \p
 /// caseInsensitive is true, then the check will be done case insensitively.
-bool isNameValid(llvm::StringRef name, bool caseInsensitiveKeywords);
+/// If \p allowEscapedName is true, IEEE-compliant escaped names are allowed to
+/// pass the check.
+bool isNameValid(llvm::StringRef name, bool caseInsensitiveKeywords,
+                 bool allowEscapedName = false);
```

Comment:
Escaped names are legal in both Verilog (1364-1995 2.7.1) and SystemVerilog (1800-2023 5.6.1). I don't see a good reason to make this optionally allowed given that (I think) all Verilog versions support escaped identifiers. The only difference is on what the identifiers are.

Reply:
> Escaped names are legal in both Verilog (1364-1995 2.7.1) and SystemVerilog (1800-2023 5.6.1). I don't see a good reason to make this optionally allowed given that (I think) all Verilog versions support escaped identifiers. The only difference is on what the identifiers are.

Yes, I concur with your viewpoint and believe that this feature should be enabled by default. Given that my understanding of CIRCT is not particularly in-depth, I am not entirely certain whether this modification might cause any side effects. Out of caution, I have temporarily set it as an optional feature, hoping that it will at least meet my needs. After extensive validation, you, I, or anyone else can set it as a default feature, which I consider to be a prudent choice.

Reply:
+1, I think default is better.